### ISSUE TITLE
repeated application of the same resource results in delete deployment

### ISSUE THREAD

**Issue by clarechu** [kubevela/kubevela#4017]
**Describe the bug**
When I updated the application resource of vela, I found that the deployment resource application was deleted by vela-core.

**To Reproduce**

```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: xxxx
  namespace: xxxx
spec:
  components:
    - name: xxxxxx
      type: webservice
      dependsOn: []
      properties:
        exposeType: NodePort
        image: xxxxxxx
        imagePullSecrets:
          - xxx-secret
        ports:
          - expose: true
            port: 80
      traits:
        - properties:
            domain: xxxxxxx
            class: nginx
            classInSpec: true
            http:
              /: 80
          type: gateway
```

1. apply application
```bash
$ kubectl apply -f demo.yaml

application.core.oam.dev/xxxx created
```

2. get pod

```bash
➜  ~ k get po -w
NAME                              READY   STATUS    RESTARTS   AGE
xxx-6cb857dc5f-nhnmt   0/1     Pending   0          0s
xxx-6cb857dc5f-nhnmt   0/1     Pending   0          0s
xxx-6cb857dc5f-nhnmt   0/1     ContainerCreating   0          0s
xxx-6cb857dc5f-nhnmt   0/1     ContainerCreating   0          3s
xxx-6cb857dc5f-nhnmt   1/1     Running             0          4s
```

3. Apply the same again yaml

```bash
kubectl apply -f demo.yaml

application.core.oam.dev/xxxx configured
```

4. get po

```bash

➜  ~ k get po -w
NAME                              READY   STATUS    RESTARTS   AGE
xxxx-6cb857dc5f-nhnmt   1/1     Terminating         0          8m35s
xxxx-6cb857dc5f-nhnmt   1/1     Terminating         0          8m36s
xxxx-6cb857dc5f-nhnmt   0/1     Terminating         0          8m37s

```

**Expected behavior**

**Screenshots**

**KubeVela Version**
vela-core : v1.3.2

**Cluster information**

Kubernetes Version [1.20.8]

**Comment by chivalryq** [comment]
This is quite strange. Is this app still reconciling? And the new Deployment haven't been created?

**Comment by clarechu** [comment]
> This is quite strange. Is this app still reconciling? And the new Deployment haven't been created?

yes， Very strange

**Comment by chivalryq** [comment]
You mentioned that the second time you apply the same application. If the app.spec are the same, kubectl report "app unchanged" instead of "app configured"

```shell
$ k apply -f app.yaml
application.core.oam.dev/app-4017 unchanged
```

Could you please check: 
1. If the second app are the same with before.
2. What minimal properties and traits in application to reproduce.

**Comment by clarechu** [comment]
The current file I did not do task changes

**Comment by chivalryq** [comment]
I'm try environment below. The second time when I apply the app.yaml kubectl just report "unchanged".

- Same vela-core version
```shell
$ vela version
CLI Version: v1.3.5
Core Version: 1.3.2
GitRevision: git-cbed2b5
GolangVersion: go1.17.10
```
- app.yaml like
```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: app-4017
spec:
  components:
    - name: comp-1
      type: webservice
      properties:
        image: nginx
        ports:
          - expose: true
            port: 80
      traits:
        - type: gateway
          properties:
            domain: example.com
            class: traefik
            http:
              /: 80
```

If you are free, please check the what are the perperties are needed in minimal to reproduce the error. That will helps a lot for tracking the issue.

**Comment by chivalryq** [comment]
About Pod restart problem, it is duplicated with https://github.com/kubevela/kubevela/issues/3751. And it has been fixed here: https://github.com/kubevela/kubevela/pull/3758 which is included in v1.3.3. You can upgrade your vela-core and retry, see if problem still exists.

If upgrade don't solve it, feel free to post more information here.

**Comment by clarechu** [comment]
@chivalryq Thank you！ I'll try the latest version first.
